mona_sdk/client.py

```diff
@@ -266,13 +224,15 @@ def __init__(
         # this point the client was successfully authenticated and self._get_user_id()
         # will work.
         self._user_id = user_id or self._get_user_id()
+
         self._rest_api_url = (
             override_rest_api_full_url
             or self._get_rest_api_export_url(override_host=override_rest_api_host)
         )
-        self._app_server_url = self._get_app_server_url(
+        self._app_server_url = override_app_server_full_url or self._get_app_server_url(
```

Comment:
why not save override_app_server_full_url in self and respect in inside _get_app_server_url? in fact - the same goes for the override_app_server_host param, no need to pass it in - currently the logic is split between here and the function itself